packages/js/product-editor/src/blocks/generic/taxonomy/use-taxonomy-search.ts

```diff
@@ -27,10 +27,10 @@ async function getTaxonomiesMissingParents(
 	} );
 	if ( missingParentIds.length > 0 ) {
 		return resolveSelect( 'core' )
-			.getEntityRecords< Taxonomy[] >( 'taxonomy', taxonomyName, {
+			.getEntityRecords( 'taxonomy', taxonomyName, {
 				include: missingParentIds,
 			} )
-			.then( ( parentTaxonomies ) => {
+			.then( ( parentTaxonomies: Taxonomy[] ) => {
 				return getTaxonomiesMissingParents(
 					[ ...( parentTaxonomies as Taxonomy[] ), ...taxonomies ],
```

Comment:
We can remove the type casting

```suggestion
					[ ...( parentTaxonomies ), ...taxonomies ],
```